exchange/bidder.go

```diff
@@ -153,6 +156,24 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb.Bi
 					}
 				}
 
+				// Only do this for request from mobile app
+				if request.App != nil {
+					for i := 0; i < len(bidResponse.Bids); i++ {
+						if bidResponse.Bids[i].BidType == openrtb_ext.BidTypeNative {
+							nativeMarkup, moreErrs := addNativeTypes(bidResponse.Bids[i].Bid, request)
+							errs = append(errs, moreErrs...)
+
+							if nativeMarkup != nil {
+								markup, err := json.Marshal(*nativeMarkup)
+								if err != nil {
+									errs = append(errs, err)
+								}
+								bidResponse.Bids[i].Bid.AdM = string(markup)
```

Comment:
If there were errors in the marshalling, do we still want to we still want to try overwriting the AdM with whatever `json.Marshal()` managed to spit out?

Reply:
Good point. `markup` very well may be `nil`, of which `string(markup)` would succeed and result in an empty string.

Reply:
Agreed. Updated